bare_metal_tests/uart/test.py

```diff
@@ -0,0 +1,282 @@
+  call host test with appropriate --baudrate parameter.
+  
+  Example usage:
+      test.py -m K64F -t GCC_ARM -b 9600
+  This command will run test on K64F platfom using GCC_ARM toolchain and
+  serial connection speed equal to 9600 b/s.
+  
+  For more options type:
+      test.py -h
+"""
+
+serial_port = None
+RETRY_LIMIT = 10
+
+# List of expected messages from the mbed device.
+message_list = ['x\0', 
+                'mbed\0',
+                'Green Tea UART bear metal test.\0', 
+                'Long Message-Long Message-Long Message-Long Message-Long Message-Long Message-Long Message-Long Message-Long Message--.\0',
+                'PASSED\0']
+
+def exit():
+    if (serial_port != None and serial_port.isOpen() == True):
+        serial_port.close()
+    sys.exit()
+        
+def run_command(cmd):
+    """! Runs command and prints proc stdout on screen.
+         It is used to build the test.
+         
+    @param cmd command to be executed.
+    @return image path.
+    """
+    image_path = str()
+    
+    # Remove build log file if exists
+    try:
+        os.remove('build.log')
+    except OSError:
+        pass
+    
+    try:
+        p = Popen(cmd, stdout=sys.stdout)
+    except OSError:
+        logging.error('Command can not be executed.')
+        exit()
+
+    # Check process status.
+    returncode = p.wait()
+    if returncode < 0:
+        logging.error('Error while executing command.')
+        exit()
+    
+    # Get image path   
+    try:
+        with open('build.log') as f:
+            build_data = json.load(f)
+            image_path = build_data['builds'][0]['bin']
+    # Image path can not be found.
+    except (KeyError, IndexError, IOError, AttributeError): 
+        # If we are here, then build process has failed.
+        logging.error('Unable to locate image file.')
+        exit()
+       
+    return image_path
+
+def build_test(target, toolchain, clean_flag, baudrate):
+    """Build the mbed test.
+    
+    @param target target id.
+    @param toolchain compile toolchain.
+    @return image path.
+    """
+    logging.info('Building mbed UART test for green tea.')
+    
+    clean = ''
+    if (clean_flag):
+        clean = '-c'
+    
+    cmd = 'mbed compile -t %s -m %s --source . --source ../.. ' \
+          '--build-data build.log %s -D BAUD_RATE=%s' \
+          % (toolchain, target, clean, baudrate)
+    
+    return run_command(cmd)
+
+def open_port(port_id, baudrate):
+    """Open serial port.
+    
+    @param port_id serial port id.
+    @param baudrate serial port baudrate.
+    """
+    global serial_port
+    try:
+        serial_port = serial.Serial(port=port_id, 
+                                    baudrate=baudrate, 
+                                    parity=serial.PARITY_NONE, 
+                                    stopbits=serial.STOPBITS_ONE,
+                                    bytesize=serial.EIGHTBITS)
+
+    except:
```

Comment:
Could you narrow what this catches? At the moment it would catch many many errors that you probably don't intend to catch.

Reply:
Specified exceptions to catch.